.circleci/config.yml

```diff
@@ -2,36 +2,88 @@
 ### This configuration file is used by CircleCI build server
 ### https://circleci.com/docs/config-sample
 ### =============================================================
-version: 2
-jobs:
-  # using tox
-  toxify:
+version: 2.1
+references:
+    tag_filter: &tag_filter
+      filters:
+        tags:
+          only: /^v\d+\.\d+\.\d+$/
+
+    reqs_print_var: &reqs_print_var
+      requires:
+        - print-env-vars
+
+    reqs_tox: &reqs_tox
+      requires:
+        - tox3-7
 
+    restore_cache: &restore_cache
+      restore_cache:
+        key: tox-{{ checksum "requirements.txt" }}-{{ checksum "requirements-dev.txt" }}-{{ checksum "tox.ini" }}
+
+jobs:
+  print-env-vars:
       docker:
         - image: python:3.8
-
       steps:
         - checkout
         - run:
             name: Print environment variables
             command: |
               echo "Parameters: FAKE_MASTER: $FAKE_MASTER"
               echo "Parameters: CONTENT_BRANCH_NAME: $CONTENT_BRANCH_NAME"
+  tox3-7:
+      docker:
+        - image: python:3.8
+      steps:
+        - checkout
+        - *restore_cache
         - run:
             name: Tox build
             command: |
               pip install tox
               tox -e py37 -v -- --cov=demisto_sdk --cov-report=html
+        - store_artifacts:
+            path: coverage_html_report
+        - persist_to_workspace:
+            root: ~/project
+            paths:
+              - coverage_html_report
+              - .coverage
+        - save_cache:
+            paths:
+              - .tox
+            key: tox-{{ checksum "requirements.txt" }}-{{ checksum "requirements-dev.txt" }}-{{ checksum "tox.ini" }}
+  tox3-8:
+      docker:
+        - image: python:3.8
+      steps:
+        - checkout
+        - run:
+            name: Tox build
+            command: |
+              pip install tox
               tox -e py38 -v
+  precommit-checks:
+      docker:
+        - image: python:3.8
+      steps:
+        - checkout
+        - *restore_cache
         - run:
             name: Pre-commit
             command: |
               . .tox/py37/bin/activate
               pre-commit --version
               pre-commit run -a
               deactivate
-        - store_artifacts:
-            path: coverage_html_report
+  coveralls-upload:
```

Comment:
why not to upload the coveralls in the tox-3-7

Reply:
That is an option that I considered last night when I was wrestling with it a bit... I did spend an inordinate amount of time last night getting it to work in a separate job though :P - but more seriously, since there are 4 concurrent jobs that are waiting on the `tox3-7` step to complete, I don't see a reason not to have that job complete sooner, and have the `coveralls` as a job that will run concurrently with those others.